<a href="https://colab.research.google.com/github/shradhyepatel/data-science-practice-log/blob/main/python/session_2_task.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### `Problem 1`: Write a program that will give you in hand monthly salary after deduction on CTC - HRA(10%), DA(5%), PF(3%) and taxes deduction as below:

> Salary(Lakhs) : Tax(%)

*   Below 5 : 0%
*   5-10 : 10%
*   10-20 : 20%
*   aboove 20 : 30%

In [4]:
from typing_extensions import ParamSpecArgs
# Write code here
# Take the Salary Input

salary = float(input("Enter your salary"))

# Calculate HRA , DA and PF by adding them all

deduction = salary * .18

new_salary = salary - deduction

# Check salary for tax deduction and calculate

if new_salary < 500000:
  pass

elif new_salary > 500000 and new_salary <1000000:
  new_salary = new_salary - (new_salary * 0.1)

elif new_salary >= 1000000 and new_salary < 2000000:
  new_salary = new_salary - (new_salary * 0.2)

else :
  new_salary = new_salary - (new_salary * 0.3)

# Print

print("In hand monthly salary : ",new_salary / 12)

Enter your salary5000000000
In hand monthly salary :  239166666.66666666


### `Problem 2`: Write a program that take a user input of three angles and will find out whether it can form a triangle or not.

In [9]:
# Write code here
# Take the input

angle1 = int(input("Enter the first angle"))
angle2 = int(input("Enter the second angle"))
angle3 = int(input("Enter the third angle"))

# Calculate and show the answers

if (angle1 + angle2 + angle3) == 180:

  if angle1 == 90 or angle2 == 90 or angle3 == 0:
    print('Its a right angle triangle')

  else:
    print("Its not a write angle triangel")

else:
  print('Its not a right angle triangle')


Enter the first angle90
Enter the second angle90
Enter the third angle0
Its a right angle triangle


### `Problem 3`: Write a program that will take user input of cost price and selling price and determines whether its a loss or a profit.

In [12]:
# Write code here
#Take the input

cost_price = int(input("Enetr the cost price"))
selling_price = int(input("ENter the selling price"))

# Calculate

if cost_price > selling_price:
  print('Loss')

elif cost_price == selling_price:
  print('No profit No Loss')

else:
  print('Profit!')


Enetr the cost price200
ENter the selling price201
Profit!


### `Problem 4`: Write a menu-driven program -
1. cm to ft
2. km to miles
3. USD to INR
4. exit

In [17]:
# Write code here
# Take Input

user_input = int(input("""
Enter 1 for cm to ft
Enter 2 for km to miles
Enter 3 to USD to INR
Enter 4 to exit : """))

if user_input == 1:
  cm_input = int(input("Enter value in cm : "))
  print("Feet : " , cm_input * 0.0328)

elif user_input == 2:
  km_input = int(input("Enter value in km"))
  print("Miles : " , km_input * 0.621)

elif user_input == 3:
  usd_input = int(input('Enter value in USD'))
  print("INR : " , usd_input * 85.87)

else:
  exit


Enter 1 for cm to ft 
Enter 2 for km to miles
Enter 3 to USD to INR
Enter 4 to exit : 3
Enter value in INR2000
INR :  171740.0


### `Problem 5` - Exercise 12: Display Fibonacci series up to 10 terms.

*Note: The Fibonacci Sequence is a series of numbers. The next number is found by adding up the two numbers before it. The first two numbers are 0 and 1. For example, 0, 1, 1, 2, 3, 5, 8, 13, 21. The next number in this series above is 13+21 = 34*

In [20]:
# Write code here

value = 0

for i in range(0,100):
  new_value = value + 1

0
1
3
6
10
15
21
28
36
45
55
66
78
91
105
120
136
153
171
190
210
231
253
276
300
325
351
378
406
435
465
496
528
561
595
630
666
703
741
780
820
861
903
946
990
1035
1081
1128
1176
1225
1275
1326
1378
1431
1485
1540
1596
1653
1711
1770
1830
1891
1953
2016
2080
2145
2211
2278
2346
2415
2485
2556
2628
2701
2775
2850
2926
3003
3081
3160
3240
3321
3403
3486
3570
3655
3741
3828
3916
4005
4095
4186
4278
4371
4465
4560
4656
4753
4851
4950


### `Problem 6` - Find the factorial of a given number.

Write a program to use the loop to find the factorial of a given number.

The factorial (symbol: `!`) means to multiply all whole numbers from the chosen number down to 1.

For example: calculate the factorial of 5

```bash
5! = 5 × 4 × 3 × 2 × 1 = 120
```

Output:

```bash
120
```

In [ ]:
# Write code here

### `Problem 7` - Reverse a given integer number.

Example:

`Input:`

```bash
76542
```

`Output:`

```bash
24567
```

In [ ]:
# Write code here

### `Problem 8`: Take a user input as integer N. Find out the sum from 1 to N. If any number if divisible by 5, then skip that number. And if the sum is greater than 300, don't need to calculate the sum further more. Print the final result. And don't use for loop to solve this problem.

**Example 1:**

`Input:`

```bash
30
```

`Output:`

```bash
276
```

In [ ]:
# Write code here

### `Problem 9`: Write a program that keeps on accepting a number from the user until the user enters Zero. Display the sum and average of all the numbers.

In [ ]:
# Write code here

###`Problem 9`: Write a program which will find all such numbers which are divisible by 7 but are not a multiple of 5, between 2000 and 3200 (both included). The numbers obtained should be printed in a comma-separated sequence on a single line.

In [ ]:
# Write code here

###`Problem 10`: Write a program, which will find all such numbers between 1000 and 3000 (both included) such that each digit of the number is an even number. The numbers obtained should be printed in a space-separated sequence on a single line.



In [ ]:
# Write code here

###`Problem 11`: A robot moves in a plane starting from the original point (0,0). The robot can move toward UP, DOWN, LEFT and RIGHT with a given steps.
The trace of robot movement is shown as the following:
```
UP 5
DOWN 3
LEFT 3
RIGHT 2
!
```
> The numbers after the direction are steps.

> `!` means robot stop there.

**Please write a program to compute the distance from current position after a sequence of movement and original point.**

*If the distance is a float, then just print the nearest integer.*

Example:

`Input`:
```
UP 5
DOWN 3
LEFT 3
RIGHT 2
!
```
`Output`:
```
2
```

In [ ]:
# Write code here

###`Problem 12`:Write a program to print whether a given number is a prime number or not

In [ ]:
# Write code here

###`Problem 13`:Print all the Armstrong numbers in a given range.
Range will be provided by the user<br>
Armstrong number is a number that is equal to the sum of cubes of its digits. For example 0, 1, 153, 370, 371 and 407 are the Armstrong numbers.

In [ ]:
# Write code here

###`Problem 14`:Calculate the angle between the hour hand and minute hand.

Note: There can be two angles between hands; we need to print a minimum of two. Also, we need to print the floor of the final result angle. For example, if the final angle is 10.61, we need to print 10.

Input:<br>
H = 9 , M = 0<br>
Output:<br>
90<br>
Explanation:<br>
The minimum angle between hour and minute
hand when the time is 9 is 90 degress.

In [ ]:
# Write code here

###`Problem 15`:Given two rectangles, find if the given two rectangles overlap or not. A rectangle is denoted by providing the x and y coordinates of two points: the left top corner and the right bottom corner of the rectangle. Two rectangles sharing a side are considered overlapping. (L1 and R1 are the extreme points of the first rectangle and L2 and R2 are the extreme points of the second rectangle).

Note: It may be assumed that the rectangles are parallel to the coordinate axis.

<img src='https://www.geeksforgeeks.org/wp-content/uploads/rectanglesOverlap.png' width='300' height='200'>

In [ ]:
# Write code here